alf/utils/common.py

```diff
@@ -803,16 +837,23 @@ def get_observation_spec():
 
     https://github.com/HorizonRobotics/alf/pull/239#issuecomment-544644558
 
+    Args:
+        field (str): a multi-step path denoted by "A.B.C".
     Returns:
         A `TensorSpec`, or a nested dict, list or tuple of
         `TensorSpec` objects, which describe the observation.
     """
     assert _env, "set a global env by `set_global_env` before using the function"
     specs = _env.observation_spec()
-    return tf.nest.map_structure(
+    specs = tf.nest.map_structure(
         lambda spec: (tf.TensorSpec(spec.shape, tf.float32)
                       if spec.dtype == tf.uint8 else spec), specs)
 
+    if field:
+        for f in field.split('.'):
```

Comment:
namedtuple cannot be indexed by a string

Reply:
actually , the spec will never be a namedtuple instance

see [`_spec_from_gym_space`  ](https://github.com/HorizonRobotics/alf/blob/5c648814b44e3a32e51a1f494964d14354976881/alf/environments/suite_gym.py#L65-L71),  it maps `gym.spaces.Tuple` to a tuple and we do not support unamed tuple observation in alf , and there is no space  `gym.spaces.UnamedTuple`

Reply:
I think the user can optionally convert a dict spec to namedtuple spec (for easier field access)? This way seems more consistent with what we have for other specs defined in ALF (they are always namedtuples).

Reply:
i think it's no necessary to convert a dict spec to namedtuple,  dict is  enough easy  for field accessing in  `observation` and `observation_space`